dynamic_rest/serializers.py

```diff
@@ -301,6 +317,29 @@ def get_name(cls):
         return cls.Meta.name
 
     @classmethod
+    def get_url(self, pk=None):
+        # if associated with a registered viewset, use its URL
+        url = getattr(self, '_url', None)
```

Comment:
Looking at the router code a little closer, I guess this _could_ be the canonical URL, but if there are multiple viewsets mapped to a single serializer, it could also be overwritten by something that's not canonical? Since canonical URLs are most useful when there is ambiguity, I think that might negate the purpose of canonical URLs and the explicit `register_resource` API.